martinLi/assignments/slidingwindow/lc53/lc53.js

```diff
@@ -1,4 +1,50 @@
+/*
+
+
+This problem requires three variables:
+window_start
+current_max
+window_end(in for loop)
+window_sum will be keeping the current sum
+window_start will be keeping track of indexes in the front of the array.
+window_end will be keeping track of indexes throughout the array.
+
+Everytime k is reached(which is when window_end reaches k) then a new current_max will be determined(or keep the old one if it is 
+  still current max)
+
+  We will then subtract the index(window_start) from window_sum.
+  window_start moves one position over.
+
+
+  Since windows sum will be subtracting the first(relative) value of the array after reaching k then this process will continue until 
+  current max is determined 
+  current max will be returned
+*/
+
 function max_subarray_size_k(k, arr) {
+  let window_start = 0, window_sum = 0, current_max = 0;
+
+  for(let window_end = 0; window_end < arr.length; window_end++){
+    window_sum += arr[window_end]
+    if(window_end >= k-1){
+      current_max = Math.max(window_sum, current_max);
+
+      window_sum -= arr[window_start]
+
+      window_start += 1;
+
+
+    }
+    return current_max;
```

Comment:
Return is situated after the if() but, NOT outside of the loop. Loop terminates the first time it runs and value returned is 0 always. Please revise and always test your code before submitting